pkg/nsx/services/vpc/vpc.go

```diff
@@ -126,6 +152,31 @@ func InitializeVPC(service common.Service) (*VPCService, error) {
 	go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeVpc, nil, VPCService.VpcStore)
 	go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeIPBlock, nil, VPCService.IpblockStore)
 
+	//initalize avi rule related store
+	if enableAviAllowRule {
+		VPCService.RuleStore = &AviRuleStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.RuleBindingType(),
+		}}
+		VPCService.GroupStore = &AviGroupStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.GroupBindingType(),
+		}}
+		VPCService.SecurityPolicyStore = &AviSecurityPolicyStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.SecurityPolicyBindingType(),
+		}}
+		VPCService.PubIpblockStore = &PubIPblockStore{ResourceStore: common.ResourceStore{
+			Indexer:     cache.NewIndexer(keyFuncAVI, nil),
+			BindingType: model.IpAddressBlockBindingType(),
+		}}
+		go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeGroup, nil, VPCService.GroupStore)
+		go VPCService.InitializeResourceStore(&wg, fatalErrors, common.ResourceTypeRule, nil, VPCService.RuleStore)
+
+		query := fmt.Sprintf("%s:%s AND visibility:EXTERNAL", common.ResourceType, common.ResourceTypeIPBlock)
+		go VPCService.PopulateResourcetoStore(&wg, fatalErrors, common.ResourceTypeIPBlock, query, VPCService.PubIpblockStore, nil)
```

Comment:
Actually I do not get why we need another pubIpBlockStore here. Could we get the external IP blocks from VPCService.IpBlockStore？